src/Esfa.Vacancy.Register.Infrastructure/Services/TrainingDetailService.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Threading.Tasks;
+using Esfa.Vacancy.Register.Domain.Entities;
+using Esfa.Vacancy.Register.Domain.Interfaces;
+using SFA.DAS.Apprenticeships.Api.Client;
+
+namespace Esfa.Vacancy.Register.Infrastructure.Services
+{
+    public class TrainingDetailService : ITrainingDetailService
+    {
+        public async Task<Framework> GetFrameworkDetails(int code)
+        {
+            using (var client = new FrameworkCodeClient())
+            {
+                var framework = await client.GetAsync(code);
+                if (framework == null) throw new TrainingResourceNotFoundException($"Framework: {code}");
+                return new Framework() { Title = framework.Title };
+            }
+        }
+
+        public async Task<Standard> GetStandardDetails(int code)
+        {
+            using (var client = new StandardApiClient())
+            {
+                var standard = await client.GetAsync(code);
+                if (standard == null) throw new TrainingResourceNotFoundException($"Standard: {code}");
```

Comment:
🤔 
I think a general System exception would suffice. The message would be different and provide context. Also it means we don't create specific exception types too often and only in cases where needed.